nifi-nar-bundles/nifi-windows-event-log-bundle/nifi-windows-event-log-processors/src/test/java/org/apache/nifi/processors/windows/event/log/EvtSubscribeTest.java

```diff
@@ -0,0 +1,299 @@
+        when(channel.getValue()).thenReturn(testChannel);
+
+        PropertyValue query = mock(PropertyValue.class);
+        when(query.getValue()).thenReturn(testQuery);
+
+        WinNT.HANDLE subscriptionHandle = mock(WinNT.HANDLE.class);
+        when(wEvtApi.EvtSubscribe(isNull(WinNT.HANDLE.class), isNull(WinNT.HANDLE.class), eq(testChannel), eq(testQuery),
+                isNull(WinNT.HANDLE.class), isNull(WinDef.PVOID.class), isA(EventSubscribeXmlRenderingCallback.class),
+                eq(WEvtApi.EvtSubscribeFlags.SUBSCRIBE_TO_FUTURE)))
+                .thenReturn(subscriptionHandle);
+
+        when(processContext.getProperty(EvtSubscribe.MAX_EVENT_QUEUE_SIZE)).thenReturn(maxEventSize);
+        when(processContext.getProperty(EvtSubscribe.MAX_BUFFER_SIZE)).thenReturn(maxBufferSize);
+        when(processContext.getProperty(EvtSubscribe.CHANNEL)).thenReturn(channel);
+        when(processContext.getProperty(EvtSubscribe.QUERY)).thenReturn(query);
+
+        evtSubscribe.subscribeToEvents(processContext);
+        ArgumentCaptor<EventSubscribeXmlRenderingCallback> callbackArgumentCaptor = ArgumentCaptor.forClass(EventSubscribeXmlRenderingCallback.class);
+        verify(wEvtApi).EvtSubscribe(isNull(WinNT.HANDLE.class), isNull(WinNT.HANDLE.class), eq(testChannel), eq(testQuery),
+                isNull(WinNT.HANDLE.class), isNull(WinDef.PVOID.class), callbackArgumentCaptor.capture(),
+                eq(WEvtApi.EvtSubscribeFlags.SUBSCRIBE_TO_FUTURE));
+
+        EventSubscribeXmlRenderingCallback callback = callbackArgumentCaptor.getValue();
+        Consumer<String> consumer = callback.getConsumer();
+        consumer.accept("TestXml1");
+        consumer.accept("TestXml2");
+        consumer.accept(testXml3);
+
+        evtSubscribe.onTrigger(processContext, processSessionFactory);
+        verify(processSession).transfer(flowFile3, EvtSubscribe.REL_SUCCESS);
+        verify(processSession, times(1)).transfer(any(FlowFile.class), eq(EvtSubscribe.REL_SUCCESS));
+        verify(processContext).yield();
+
+        assertEquals(processSessionFactory, evtSubscribe.getProcessSessionFactory());
+        assertEquals(MediaType.APPLICATION_XML_UTF_8.toString(), mimeType1.get());
+        assertEquals(testXml3, Charsets.UTF_8.decode(ByteBuffer.wrap(byteArrayOutputStream1.toByteArray())).toString());
+
+        consumer.accept(testXml4);
+        assertEquals(MediaType.APPLICATION_XML_UTF_8.toString(), mimeType2.get());
+        assertEquals(testXml4, Charsets.UTF_8.decode(ByteBuffer.wrap(byteArrayOutputStream2.toByteArray())).toString());
+
+        verify(processSession).transfer(flowFile6, EvtSubscribe.REL_SUCCESS);
+        verify(processSession, times(2)).transfer(any(FlowFile.class), eq(EvtSubscribe.REL_SUCCESS));
+
+        evtSubscribe.closeSubscriptionHandle();
+        verify(kernel32).CloseHandle(subscriptionHandle);
+    }
+
+    @Test
+    public void testLifecycle() throws InvocationTargetException, IllegalAccessException {
```

Comment:
Unit tests should be targeted to test specific things so that when they fail the developer knows exactly what is failing. This test combines many different things which should be broken out into their own unit test. For stuff that happens before each test (mocking or initializing things) use the @Before annotation.